bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ThingLinkManager.java

```diff
@@ -67,11 +69,11 @@
 
     private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(THREADPOOL_NAME);
 
-    private ThingRegistry thingRegistry;
-    private ManagedThingProvider managedThingProvider;
-    private ItemRegistry itemRegistry;
-    private ItemChannelLinkRegistry itemChannelLinkRegistry;
-    private ChannelTypeRegistry channelTypeRegistry;
+    private @NonNullByDefault({}) ThingRegistry thingRegistry;
```

Comment:
Wouldn't it make sense to use constructor injection and ensure that every instance got a non null thing registry (and the other below)?

I agree that such a change can be separated, but perhaps we can do it before and so not need to add non null by default exclusions.

Reply:
Can also be done later...

Reply:
I thought about that too. But it is more clear if we have it in a separate PR.